### PR TITLE
GEODE-6748: First part of solution

### PR DESCRIPTION
**Pull request by mivanac**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [*] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [*] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [*] Is your initial contribution a single, squashed commit?

- [*] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by mivanac** [comment]
Do not review. WIP

**Comment by mhansonp** [comment]
Hi @mivanac,

I was not sure if this PR is ready for review, at your leisure, please send a note to the dev@geode.apache.org requesting a review. I would have done it when I went through the list of PRs this morning, but I was not 100% sure. Thanks for all your efforts.

Thanks,
Mark

File: geode-core/src/main/java/org/apache/geode/cache/client/internal/InvalidateOp.java [hunk 1]
```diff
@@ -39,11 +47,38 @@
...
+    if (prSingleHopEnabled) {
+      ClientMetadataService cms = region.getCache().getClientMetadataService();
+      ServerLocation server =
+          cms.getBucketServerLocation(region, Operation.INVALIDATE, event.getKey(), null,
+              event.getCallbackArgument());
+      if (server != null) {
+        try {
+          PoolImpl poolImpl = (PoolImpl) pool;
+          boolean onlyUseExistingCnx = ((poolImpl.getMaxConnections() != -1
+              && poolImpl.getConnectionCount() >= poolImpl.getMaxConnections()) ? true : false);
```
Comment (jmelchio) [inline geode-core/src/main/java/org/apache/geode/cache/client/internal/InvalidateOp.java]: I think you can omit the `? true : false` as the preceding expression will already resolve to one or the other.

Comment (mivanac) [reply geode-core/src/main/java/org/apache/geode/cache/client/internal/InvalidateOp.java]: Thanks for review.

**Pull request merged**